demo/image_demo.py

```diff
@@ -1,94 +1,104 @@
 # Copyright (c) OpenMMLab. All rights reserved.
-import asyncio
+import warnings
 from argparse import ArgumentParser
 
-import mmcv
+from mmengine.logging import print_log
 
-from mmdet.apis import (async_inference_detector, inference_detector,
-                        init_detector)
-from mmdet.registry import VISUALIZERS
+from mmdet.apis import DetInferencer
 
 
 def parse_args():
     parser = ArgumentParser()
-    parser.add_argument('img', help='Image file')
-    parser.add_argument('config', help='Config file')
-    parser.add_argument('checkpoint', help='Checkpoint file')
-    parser.add_argument('--out-file', default=None, help='Path to output file')
+    parser.add_argument(
+        'inputs', type=str, help='Input image file or folder path.')
+    parser.add_argument(
+        'model',
+        type=str,
+        help='Config or checkpoint .pth file or the model name defined '
+        'in metafile. The model configuration file will try to read '
+        'from .pth if the parameter is a .pth weights file.')
+    parser.add_argument('--weights', default=None, help='Checkpoint file')
+    parser.add_argument(
+        '--img-out-dir',
+        type=str,
+        default='outputs',
+        help='Output directory of images.')
     parser.add_argument(
         '--device', default='cuda:0', help='Device used for inference')
+    parser.add_argument(
+        '--pred-score-thr',
+        type=float,
+        default=0.3,
+        help='bbox score threshold')
+    parser.add_argument(
+        '--batch-size', type=int, default=1, help='Inference batch size.')
+    parser.add_argument(
+        '--show',
+        action='store_true',
+        help='Display the image in a popup window.')
+    parser.add_argument(
+        '--no-save-image',
+        action='store_true',
+        help='Do not save detection vis results')
+    parser.add_argument(
+        '--print-result',
+        action='store_true',
+        help='Whether to print the results.')
+    parser.add_argument(
+        '--pred-out-file',
```

Comment:
If the `inputs` is an image folder, will the dumped json file be a list of results?